planning/autoware_static_centerline_generator/src/centerline_source/optimization_trajectory_based_centerline.cpp

```diff
@@ -188,4 +277,112 @@ std::vector<TrajectoryPoint> OptimizationTrajectoryBasedCenterline::optimize_tra
 
   return whole_optimized_traj_points;
 }
+
+std::shared_ptr<autoware_planning_msgs::msg::LaneletRoute_<std::allocator<void>>>
+OptimizationTrajectoryBasedCenterline::createRoutePtr(
+  const Path & raw_path, std::shared_ptr<RouteHandler> & route_handler_ptr_) const
+{
+  // create route_ptr_
+  std::shared_ptr<autoware_planning_msgs::msg::LaneletRoute_<std::allocator<void>>> route_ptr_ =
+    std::make_shared<autoware_planning_msgs::msg::LaneletRoute_<std::allocator<void>>>();
+
+  lanelet::ConstLanelets all_lanelets;
+
+  for (size_t i = 1; i < raw_path.points.size(); i++) {
+    const auto start_point = raw_path.points.at(i - 1);
+    const auto goal_point = raw_path.points.at(i);
+    // std::cerr << "iketa goal_pose" << goal_point.pose.position.x << std::endl;
+    lanelet::ConstLanelets path_lanelets;
+    if (!route_handler_ptr_->planPathLaneletsBetweenCheckpoints(
+          start_point.pose, goal_point.pose, &path_lanelets)) {
+      // RCLCPP_WARN(node.get_logger(), "Failed to plan route.");
+    }
+
+    for (const auto & lane : path_lanelets) {
+      if (!all_lanelets.empty() && lane.id() == all_lanelets.back().id()) continue;
+      all_lanelets.push_back(lane);
+    }
+  }
+
+  route_handler_ptr_->setRouteLanelets(all_lanelets);
+  std::vector<autoware_planning_msgs::msg::LaneletSegment> route_sections =
+    route_handler_ptr_->createMapSegments(all_lanelets);
+
+  // set nessary data
+  route_ptr_->header.frame_id = "map";
+  route_ptr_->set__start_pose(start_pose);
+  route_ptr_->set__goal_pose(goal_pose);
+  route_ptr_->set__segments(route_sections);
+  return route_ptr_;
+}
+
+std::shared_ptr<autoware::path_generator::PathGenerator>
+OptimizationTrajectoryBasedCenterline::createPathGeneratorNode(
+  const geometry_msgs::msg::Pose current_pose, LaneletMapBin::ConstSharedPtr & map_bin_ptr_,
+  std::shared_ptr<autoware_planning_msgs::msg::LaneletRoute_<std::allocator<void>>> & route_ptr_)
+  const
+{
+  // create path_generator_node
+  const auto path_generator_dir =
+    ament_index_cpp::get_package_share_directory("autoware_path_generator");
+  const auto node_options = rclcpp::NodeOptions{}.arguments(
+    {"--ros-args", "--params-file", path_generator_dir + "/config/path_generator.param.yaml"});
+  const auto path_generator_node =
+    std::make_shared<autoware::path_generator::PathGenerator>(node_options);
+
+  // set data in path_generator
+  autoware::path_generator::PathGenerator::InputData path_generator_input;
+  path_generator_input.lanelet_map_bin_ptr = map_bin_ptr_;
+  path_generator_input.odometry_ptr = utils::convert_to_odometry(current_pose);
+  path_generator_input.route_ptr = route_ptr_;
+  path_generator_node->set_planner_data(path_generator_input);
+
+  return path_generator_node;
+}
+
+std::shared_ptr<autoware::behavior_path_planner::PlannerData>
+OptimizationTrajectoryBasedCenterline::createPlannerData(
+  std::shared_ptr<RouteHandler> & route_handler_ptr_, path_generator::Params & params,
```

Comment:
```suggestion
  std::shared_ptr<RouteHandler> & route_handler_ptr, path_generator::Params & params,
```

`_` is for member variable